### PR TITLE
fix(nvim_lsp): attempt to index local result L60

### PR DESCRIPTION
**Pull request by tami5**
possible fix of #2 . After adding those lines and few test, I no longer have the aforementioned error message. Not sure if its the right solution to the issue.

### CODE REVIEW

File: lua/compe_nvim_lsp/source.lua [hunk 1]
```diff
@@ -47,18 +49,32 @@ end
...
   local params = vim.lsp.util.make_position_params()
+  local buf_has_clients = function()
+    for _ in pairs(vim.lsp.buf_get_clients()) do
+      return true
+    end
+    return false
+  end
+
+  if not buf_has_clients() then
+    Source.callback = true
```
Comment (hrsh7th) [inline lua/compe_nvim_lsp/source.lua:60]: You should call `return args.abort()` and remove `Source.callback` property.

File: lua/compe_nvim_lsp/source.lua [hunk 2]
```diff
@@ -47,18 +49,32 @@ end
...
   params.context = {
     triggerKind = (args.trigger_character_offset > 0 and 2 or (args.incomplete and 3 or 1))
   }
   if args.trigger_character_offset > 0 then
     params.context.triggerCharacter = args.context.before_char
   end
 
-  self.client.request('textDocument/completion', params, function(method, err, result)
+  self.client.request('textDocument/completion', params, function(err, _, result)
+    if err or not result then Source.callback = true return end
```
Comment (hrsh7th) [inline lua/compe_nvim_lsp/source.lua:72]: Same.

**Comment by hrsh7th** [comment]
Thank you.
I've added few comments.

File: lua/compe_nvim_lsp/source.lua [hunk 3]
```diff
@@ -47,18 +49,32 @@ end
 
 function Source:complete(args)
   local params = vim.lsp.util.make_position_params()
+  local buf_has_clients = function()
```
Comment (hrsh7th) [inline lua/compe_nvim_lsp/source.lua:52]: I think should be good to change to `if vim.lsp.client_is_stopped(self.client.id) then return args.abort() end`.

**Comment by tami5** [comment]
Done

**Comment by hrsh7th** [comment]
LGTM. Thank you!

**Comment by tami5** [comment]
No thank you @hrsh7th, i'm so amazed of how awesome it feels using it. You did a great job honestly

**Pull request merged**